geniza/annotations/models.py

```diff
@@ -247,6 +249,14 @@ def sanitize_html(cls, html):
         else:
             return cleaned_html
 
+    @property
+    def etag(self):
+        """Compute and return an md5 hash of content to use as an ETag"""
+        # must be a string encoded as utf-8 to compute md5 hash
+        content_str = json.dumps(self.content, sort_keys=True).encode("utf-8")
+        # ETag should be wrapped in double quotes, per Django @condition docs
+        return f'"{hashlib.md5(content_str).hexdigest()}"'
```

Comment:
I didn't remember this and it isn't obvious to me from the docs that it needs to be wrapped this way. I would have thought the django middleware would generate the header as needed. I presume you have tested this and it does the right thing?

Reply:
I'm going off the decorator's docstring (which appears on hover in my editor)—and you're right! Looks like Django would wrap it anyway, but I suppose we'll save the tiny amount of computation required to do that.

https://github.com/django/django/blob/ae10146793dca0e0594c7acdee20ca3810983f39/django/views/decorators/http.py#L94-L97

Reply:
It's more the readability of the code than the computation. Thanks for checking!